### PR TITLE
Firefox: Update to version 89.0. [ci skip]

### PR DESCRIPTION
**Pull request by eoli3n**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Local build results
- [x] building for glibc x86_64

### CODE REVIEW

**Comment by Duncaen** [comment]
Works for `x86_64-glibc`, currently building `x86_64-musl`.

**Comment by ericonr** [comment]
Don't forget `firefox-i18n`.

**Comment by eoli3n** [comment]
Fixed ``disable-minidump.patch`` but didn't test yet.

**Comment by eoli3n** [comment]
@Duncaen ``./xbps-src patch firefox`` succeed, would you like to rebuild with current PR state ?

**Comment by Duncaen** [comment]
`x86_64-musl` finished and works in a very basic open and play a video on youtube test.

**Comment by ericonr** [comment]
`x86_64-musl` working for basic usage, but localization stopped working via the system package and I downloaded language packs via `Settings`. I wonder if they changed the build option or removed it entirely.

File: srcpkgs/firefox/patches/disable-minidump.patch [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
    "toolkit/library/rust/",
-   "toolkit/mozapps/defaultagent/rust",
-@@ -79,7 +78,6 @@ spirv_cross = { git = "https://github.com/kvark/spirv_cross", branch = "wgpu4" }
+@@ -85,7 +84,6 @@ spirv_cross = { git = "https://github.com/kvark/spirv_cross", branch = "wgpu5" }
  # failure's backtrace feature might break our builds, see bug 1608157.
  failure = { git = "https://github.com/badboy/failure", rev = "64af847bc5fdcb6d2438bec8a6030812a80519a5" }
  failure_derive = { git = "https://github.com/badboy/failure", rev = "64af847bc5fdcb6d2438bec8a6030812a80519a5" }
--minidump_writer_linux = { git = "https://github.com/msirringhaus/minidump_writer_linux.git", rev = "01c7a0da8d34059f7dae8ab9e7512529ff16347a" }
- 
+-minidump_writer_linux = { git = "https://github.com/msirringhaus/minidump_writer_linux.git", rev = "667c38bf1d9a8e$
```
Comment (Duncaen) [inline srcpkgs/firefox/patches/disable-minidump.patch:20]: broken patch

**Comment by Duncaen** [comment]
`disable-minidump.patch` not required since `--disable-crashreporter` now disables it correctly: https://bugzilla.mozilla.org/show_bug.cgi?id=1701623.

File: srcpkgs/firefox/patches/disable-minidump.patch (hunk 1 above)
Comment (eoli3n) [reply srcpkgs/firefox/patches/disable-minidump.patch]: fixed the fixed patch

**Comment by eoli3n** [comment]
And just removed the double fixed patch ``disable-minidump.patch``

**Comment by eoli3n** [comment]
I confirm that firefox-i18n-fr does not work anymore here.

**Comment by ericonr** [comment]
https://searchfox.org/mozilla-central/source/toolkit/mozapps/extensions/internal/AddonSettings.jsm the option still exists, so I don't know what broke. Maybe they changed the search paths?

**Comment by eoli3n** [comment]
@ericonr restarting firefox multiple times make it work, after 2 restart, firefox is in ``fr`` for me

**Pull request closed without merging**